### PR TITLE
Migrate reopen unit tests

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: test/byContract/IexecPoco/05_reopen.test.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        await matchOrdersAndInitializeTask(3);
+        await contribute(worker1, resultDigest);
+        await contribute(worker2, resultDigest);
+        // Consensus reached, reveal for worker 1.
+        await iexecPoco
+            .connect(worker1)
+            .reveal(taskId, resultDigest)
+            .then((tx) => tx.wait());
+        const task = await iexecPoco.viewTask(taskId);
+        console.log('🚀 ~ it ~ task:', task.status);
```
Comment (zguesmi) [inline test/byContract/IexecPoco/05_reopen.test.ts:207]:
```suggestion
```

File: test/byContract/IexecPoco/05_reopen.test.ts [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        }
+        const taskBefore = await iexecPoco.viewTask(taskId);
+        expect(taskBefore.status).to.equal(TaskStatusEnum.REVEALING);
+        expect(taskBefore.consensusValue).to.equal(resultHash);
+        expect(taskBefore.revealCounter).to.equal(0);
+        expect(taskBefore.winnerCounter).to.equal(3);
+        // Time travel beyond reveal deadline but before final deadline.
+        await setNextBlockTimestamp(taskBefore.revealDeadline).then(() => mine());
+        expect(taskBefore.status).to.equal(TaskStatusEnum.REVEALING); // require 1
+        expect(await time.latest()).to.be.lessThan(taskBefore.finalDeadline); // require 2
```
Comment (jeremyjams) [inline test/byContract/IexecPoco/05_reopen.test.ts:103]: Use single latest call?

**Review by gfournieriExec — APPROVED** [review]

File: test/byContract/IexecPoco/05_reopen.test.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        worker3,
+        worker4,
+        anyone,
+    ]: SignerWithAddress[] = [];
+    let ordersAssets: OrdersAssets;
+    let ordersPrices: OrdersPrices;
+    let [dealId, taskId]: string[] = [];
+    let taskIndex: number;
+
+    beforeEach('Deploy, match orders, and contribute', async () => {
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/byContract/IexecPoco/05_reopen.test.ts:48]: To give an idea about common setup of all tests.

File: test/byContract/IexecPoco/05_reopen.test.ts (hunk 2 above)
Comment (zguesmi) [reply test/byContract/IexecPoco/05_reopen.test.ts]: Done [Apply review suggestions](https://github.com/iExecBlockchainComputing/PoCo/pull/133/commits/433ccd10820549c79ed9844b0d14eea6127b4593)

File: test/byContract/IexecPoco/05_reopen.test.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        // Only 1 contribution, consensus not reached yet.
+        await contribute(worker1, resultDigest);
+        const task = await iexecPoco.viewTask(taskId);
+        expect(task.status).to.be.lessThan(TaskStatusEnum.REVEALING); // require 1 <--
+        expect(await time.latest()).to.be.lessThan(task.finalDeadline); // require 2
+        expect(task.revealCounter).to.equal(0); // require 3.2
+        await expect(iexecPoco.reopen(taskId)).to.revertedWithoutReason();
+    });
+
+    it('Should not reopen task when is after revealing', async () => {
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/byContract/IexecPoco/05_reopen.test.ts]: Done [Apply review suggestions](https://github.com/iExecBlockchainComputing/PoCo/pull/133/commits/433ccd10820549c79ed9844b0d14eea6127b4593)

File: test/byContract/IexecPoco/05_reopen.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply test/byContract/IexecPoco/05_reopen.test.ts:48]: Yes but here it's only deploy ? I mean I don't see match orders / contribute in the before each => I see that all test are indeed doing this setup thought

[the remark this answers is not in the document]
Comment (zguesmi) [reply test/byContract/IexecPoco/05_reopen.test.ts]: My bad, It was like that at the beginning. Fixed thanks !

**Pull request merged**